### PR TITLE
Increase glTF parsing performance

### PR DESCRIPTION
**Pull request by qmuntal**
First of all, great project!

While testing your usage of [qmuntal/gltf](https://github.com/qmuntal/gltf) I noticed `LoadGLTFData` consumes quite a lot of memory compared to the size of the parsed glTF file. For example, `./example/logo/tetra3d.glb` weights ~58KB and `LoadGLTFData` consumes ~3MB and produces ~27k allocations!

My take is that most of the memory consumed by that function is accidental. A sane size to in-memory ratio would be 2 or 3x, not 58x.

This PR contains several memory improvements, which as side effect also have some net speed gains:

```txt
C:\Users\qmuntaldiaz\code\Tetra3d>benchstat old.txt new.txt
name             old time/op    new time/op    delta
LoadGLTFData-12    1.97ms ±10%    1.36ms ±15%  -30.67%  (p=0.000 n=9+9)

name             old speed      new speed      delta
LoadGLTFData-12  29.9MB/s ± 9%  43.1MB/s ±13%  +44.27%  (p=0.000 n=9+9)

name             old B/op       new B/op       delta
LoadGLTFData-12    3.17MB ± 0%    1.64MB ± 0%  -48.19%  (p=0.000 n=10+10)

name             old allocs/op  new allocs/op  delta
LoadGLTFData-12     27.6k ± 0%     25.5k ± 0%   -7.60%  (p=0.000 n=10+9)
```

### CODE REVIEW

File: mesh.go [hunk 1]
```diff
@@ -457,13 +458,17 @@ func NewMeshPart(mesh *Mesh, material *Material) *MeshPart {
...
+		Material:        part.Material,
+	}
+	for i, tri := range part.Triangles {
 		newTri := tri.Clone()
-		newMP.Triangles = append(newMP.Triangles, newTri)
-		newMP.sortedTriangles = append(newMP.sortedTriangles, newTri)
-		newMP.Vertices = append(newMP.Vertices, tri.Vertices...)
+		newMP.Triangles[i] = newTri
+		newMP.sortedTriangles[i] = newTri
+		newMP.Vertices = tri.Vertices
```
Comment (SolarLune) [inline mesh.go:471]: Shouldn't this still be newMP.Vertices = append(newMP.Vertices, tri.Vertices...)? Here the mesh part (which is a collection of triangles / vertices under a single material) is directly setting its vertex slice to be a reference to a triangle's vertex list (it needs to be all of them).

**Comment by SolarLune** [comment]
Oh, wow, thanks a lot for the help - this much of a massive memory savings is great to see. I've reviewed the PR and made a comment - thanks a lot, again for submitting it!

File: mesh.go (hunk 1 above)
Comment (qmuntal) [reply mesh.go:471]: Wops, good catch. I'll commit a fix.

Comment (SolarLune) [reply mesh.go]: Great - thanks a lot for the fix!

**Comment by SolarLune** [comment]
Fantastic. Thank you very much for the PR, again!

**Pull request merged**